### PR TITLE
docs: functions in `pickers.lua`

### PR DESCRIPTION
**Pull request by l-kershaw**
This is almost entirely copied from #927, but that PR was so out of date it was easier to just make a new one than to deal with the merge conflicts 😅 

I decided it wasn't worth doing docs for `entry_manager.lua` and `linked_list.lua` at the moment, since there have been discussions about reimplementing things in that area.

### CODE REVIEW

File: lua/telescope/pickers.lua [hunk 1]
```diff
@@ -191,12 +193,17 @@ function Picker:get_reset_row()
   end
 end
 
+--- Check if the picker is no longer in use
+---@return boolean/nil: `true` if picker is closed, `nil` otherwise
```
Comment (Conni2461) [inline lua/telescope/pickers.lua:197]: its `|` not `/`

File: lua/telescope/pickers.lua [hunk 2]
```diff
@@ -1210,6 +1293,11 @@ function Picker:_do_selection(prompt)
   end
 end
 
+--- Wrapper function for `Picker:new` that incorporates user provided `opts`
+--- with the telescope `defaults`
+---@param opts table
+---@param defaults table
+---@return any
```
Comment (Conni2461) [inline lua/telescope/pickers.lua:1300]: There are technically class annotation that you can do and then you can say `---@return Picker` and we can document fields for that class and so much more but i think that works to :laughing:

**Comment by Conni2461** [comment]
If i were you i would merge this before tj comes back and rewrites pickers again: https://github.com/nvim-telescope/telescope.nvim/pull/1491

and let him update documentation :rofl:

**Comment by l-kershaw** [comment]
Agreed, best to get this merged before tj rewrites it all again 🤣

I'll merge now, and if there's anything we want to add/change later, we can just do another PR 🙂

**Pull request merged**